File: src/common/models/ipmodels/ipblock.go
package ipmodels

import (
	"time"

	ipapisdk "github.com/phoenixnap/go-sdk-bmc/ipapi/v2"
)

type IpBlock struct {
	Id                   string                   `yaml:"id" json:"id"`
	Location             string                   `yaml:"location" json:"location"`
	CidrBlockSize        string                   `yaml:"cidrBlockSize" json:"cidrBlockSize"`
	Cidr                 string                   `yaml:"cidr" json:"cidr"`
	Status               string                   `yaml:"status" json:"status"`
	AssignedResourceId   *string                  `yaml:"assignedResourceId,omitempty" json:"assignedResourceId,omitempty"`
	AssignedResourceType *string                  `yaml:"assignedResourceType,omitempty" json:"assignedResourceType,omitempty"`
	Description          *string                  `yaml:"description,omitempty" json:"description,omitempty"`
	Tags                 []ipapisdk.TagAssignment `json:"tags,omitempty"`
	IsBringYourOwn       bool                     `json:"isBringYourOwn"`
	CreatedOn            time.Time                `json:"createdOn"`
}

func IpBlockFromSdk(ipBlock ipapisdk.IpBlock) IpBlock {
	return IpBlock{
		Id:                   ipBlock.Id,
		Location:             ipBlock.Location,
		CidrBlockSize:        ipBlock.CidrBlockSize,
		Cidr:                 ipBlock.Cidr,
		Status:               ipBlock.Status,
		AssignedResourceId:   ipBlock.AssignedResourceId,
		AssignedResourceType: ipBlock.AssignedResourceType,
		Description:          ipBlock.Description,
		Tags:                 ipBlock.Tags,
	}
}

File: e2e-tests/src/test/java/com/pnapctl/tests/datasources/models/ipblock/IpBlock.java
package com.pnapctl.tests.datasources.models.ipblock;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.pnapctl.tests.datasources.models.tag.TagAssignment;

@JsonIgnoreProperties(ignoreUnknown = true)
public record IpBlock(
        String id,
        String location,
        String cidrBlockSize,
        String cidr,
        String ipVersion,
        String status,
        String parentIpBlockAllocationId,
        String assignedResourceId,
        String assignedResourceType,
        String description,
        List<TagAssignment> tags,
        Boolean isSystemManaged,
        boolean isBringYourOwn,
        String createdOn) {

}
